seacrowd/sea_datasets/sea_wiki/sea_wiki.py

```diff
@@ -0,0 +1,219 @@
+
+            elif "seacrowd" in _config_schema_name:
+                features = schemas.ssp_features
+
+            else:
+                raise ValueError(f"Unexpected schema received! {_config_schema_name}")
+
+        # summarization schema
+        elif CONFIG_SUFFIXES_FOR_TASK[1] in _config_schema_name:
+            if "source" in _config_schema_name:
+                features = datasets.Features({"url": datasets.Value("string"), "title": datasets.Value("string"), "text": datasets.Value("string")})
+
+            elif "seacrowd" in _config_schema_name:
+                features = schemas.text2text_features
+
+            else:
+                raise ValueError(f"Unexpected schema received! {_config_schema_name}")
+
+        else:
+            raise ValueError(f"Received unexpected config schema of {_config_schema_name}!")
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: DownloadManager) -> List[datasets.SplitGenerator]:
+        # args of dl_manager is a placeholder since this data loader will wrap the hf `load_dataset` from given _URL
+        # directly using `_load_hf_data_from_remote`
+        return [datasets.SplitGenerator(name=datasets.Split.TRAIN)]
+
+    def _load_hf_data_from_remote(self):
+        # construct remote_hf_reference by the last 2 of string-spliited of "/"
+        _remote_hf_reference = "/".join(_URL.split("/")[-2:])
+        _lang_args = _LANG_CONFIG[self.config.subset_id]["source_subset"]
+        _split = "train"
+
+        logger.info(f"Loading dataset from remote HF {_remote_hf_reference} with seacrowd lang args of {self.config.subset_id} and source lang args of {_lang_args} and split args of {_split}")
+        _hf_dataset_source = load_dataset(_remote_hf_reference, lang=_lang_args, split=_split)
+
+        return _hf_dataset_source
+
+    def _generate_examples(self) -> Tuple[int, Dict]:
+
+        _config_schema_name = self.config.schema
+        loaded_data = self._load_hf_data_from_remote()
+
+        # iterate over datapoints and arrange hf dataset schema in source to match w/ config args:
+        for id_, _data in enumerate(loaded_data):
+            if "source" in _config_schema_name:
+                yield id_, {colname: _data[colname] for colname in self.info.features}
+
+            # for ssp schema
+            elif "seacrowd" in _config_schema_name and CONFIG_SUFFIXES_FOR_TASK[0] in _config_schema_name:
+                yield id_, {"id": id_, "text": _data["text"]}
+
+            # for summary schema
+            elif "seacrowd" in _config_schema_name and CONFIG_SUFFIXES_FOR_TASK[1] in _config_schema_name:
+                yield id_, {"id": id_, "text_1": _data["text"], "text_2": _data["title"], "text_1_name": "document", "text_2_name": "title"}
```

Comment:
Since `"text_1_name": "document", "text_2_name": "title"` is hardcoded, why can't we write `yield id_, {"id": id_, "document": _data["text"], "title": _data["title"]}` instead? Maybe I am missing something here (potential conflicts with line 208?) or perhaps there's foreseeable extension of this schema where `text_1_name` would refer to other document type?